consensus/src/consensus/mod.rs

```diff
@@ -610,6 +615,97 @@ impl ConsensusApi for Consensus {
         sample_headers
     }
 
+    fn get_utxo_return_script_public_key(&self, txid: Hash, target_daa_score: u64) -> Option<ScriptPublicKey> {
```

Comment:
Instead of an Option<ScriptPublicKey>, i think an enum with background on the search result, makes more sense, or alternatively in "not found" cases differentiated consensus errors, which can be propagated back to rpc.  I see three distinct reasons why a script_public_key may not be found, amounting to:

1) Data is pruned, 
2) Txid is not found at the target_daa_score, 
3) Searched txid is a coinbase tx.

Reply:
Updated to return an enum